common/src/View/FaceTool.cpp

```diff
@@ -53,6 +56,29 @@ namespace TrenchBroom {
             }
             return MR_Deny;
         }
+
+        bool FaceTool::canDoCsgConvexMerge() {
+            return handleManager().selectedHandleCount() > 1;
+        }
+
+        void FaceTool::csgConvexMerge() {
+            std::vector<vm::vec3> vertices;
+            const auto faces = handleManager().selectedHandles();
+            vm::polygon3::getVertices(std::begin(faces), std::end(faces), std::back_inserter(vertices));
+            const Polyhedron3 polyhedron(vertices);
+            if (!polyhedron.polyhedron() || !polyhedron.closed()) {
+                return;
+            }
+
+            MapDocumentSPtr document = lock(m_document);
```

Comment:
This logic is not duplicated across all vertex tools, I would suggest to have a template method in `VertexToolBase` that calls a method to get the vertices of the selected handles.

Same for `canDoCsgConvexMerge`, maybe that can be entirely implemented in the base class.